com.unity.render-pipelines.universal/Tests/Runtime/Light2DTests.cs

```diff
@@ -96,5 +101,77 @@ public void LightIsNotInVisibleListIfNotInCameraView()
 
             Assert.IsFalse(cullResult.visibleLights.Contains(light));
         }
+
+        [Test]
+        public void CachedMeshDataIsUpdatedOnChange()
+        {
+            var shapePath = new Vector3[4] { new Vector3( 0, 0, 0), new Vector3(1,0,0), new Vector3(1, 1, 0), new Vector3(0, 1, 0) };
+            var cachedMesh = m_TestObjectCached.AddComponent<LightCachedMeshData>();
+            var light = m_TestObjectCached.AddComponent<Light2D>();
+            light.lightType = Light2D.LightType.Freeform;
+
+            cachedMesh.allowUpdateOnTests = true;
+            light.SetShapePath(shapePath);
+            light.UpdateMesh(true);
+
+            Assert.AreEqual(true, cachedMesh.HasGeometryData() );
+        }
+
+        [UnityTest]
+        public IEnumerator CachedMeshDataIsUsedOnRuntime()
+        {
+            var shapePath = new Vector3[4] { new Vector3( 0, 0, 0), new Vector3(1,0,0), new Vector3(1, 1, 0), new Vector3(0, 1, 0) };
+            var cachedMesh = m_TestObjectCached.AddComponent<LightCachedMeshData>();
+            var light = m_TestObjectCached.AddComponent<Light2D>();
+            light.lightType = Light2D.LightType.Freeform;
+            cachedMesh.allowUpdateOnTests = true;
+            light.SetShapePath(shapePath);
+            light.UpdateMesh(true);
+
+            yield return new EnterPlayMode();
```

Comment:
I think all tests in this files are Play Mode tests? Why do we need this?

Reply:
Because we need to check Behavior of GeometryCache on Runtime. When it uses and when it skips it ?

Reply:
Fixed as below.